addon/services/viewport.js

```diff
@@ -66,5 +70,29 @@ export default Service.extend({
     if (this._globalWatcher) {
       this._globalWatcher.destroy();
     }
+  },
+
+  invalidate() {
+    spaniel.invalidate();
+  },
+
+  on(eventName, callback, root = window) {
+    root.addEventListener(eventName, callback, false);
+  },
+
+  off(eventName, callback, root = window) {
+    root.removeEventListener(eventName, callback, false);
+  },
+
+  onDebounce(eventName, callback, interval, root = window) {
```

Comment:
there is already a run.debounce util in ember itself. also, this doesn't not use anything in the service and does not seem like it has to do with spaniel.